src/main/java/database/StorageDealerImpl.java

```diff
@@ -0,0 +1,21 @@
+package database;
+
+import java.util.Map;
+
+public class StorageDealerImpl implements StorageDealer {
+    @Override
+    public void checkBalance(Map<String, Integer> transactions) {
+        transactions.entrySet().stream()
+                .filter(entry -> entry.getValue() < 0)
+                .findAny()
+                .ifPresent(entry -> {
+                    throw new RuntimeException("Negative balance for fruit: " + entry.getKey()
+                            + " with quantity: " + entry.getValue());
+                });
+    }
```

Comment:
It's a code quality issue. It might be better to use `forEach` instead of `findAny` and `ifPresent` here. The name `findAny` implies that you are looking for an item, but not necessarily going to do anything with it. In this case, you are not just finding an item, but throwing an exception if it exists. So `forEach` would be more appropriate here.